src/main/java/uk/gov/hmcts/reform/juddata/camel/validator/JsrValidatorInitializer.java

```diff
@@ -146,6 +147,47 @@ public void setValues(PreparedStatement ps, ConstraintViolation<T> argument) thr
         log.info("::::JsrValidatorInitializer data processing audit complete::::");
     }
 
+
+    /**
+     * Auditing JSR Exception for skipped parent in child.
+     *
+     * @param keys List
+     * @param exchange Exchange
+     *
+     */
+    public void auditJsrExceptions(List<String> keys, String fieldInError, Exchange exchange) {
+
+        log.info("::::JsrValidatorInitializer data processing audit start for skipping parent table violation::::");
+
+        DefaultTransactionDefinition def = new DefaultTransactionDefinition();
+        def.setName("Jsr exception logs");
+        def.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRED);
+        RouteProperties routeProperties = (RouteProperties) exchange.getIn().getHeader(ROUTE_DETAILS);
+        Map<String, String> globalOptions = camelContext.getGlobalOptions();
+        String schedulerTime = globalOptions.get(SCHEDULER_START_TIME);
+
+        jdbcTemplate.batchUpdate(
+                invalidJsrSql,
+                keys,
+                10,
+                new ParameterizedPreparedStatementSetter<String>() {
```

Comment:
please move batch size property to application.yaml
And batch size can be incrreased.. 10 seems to be too small for batch

Reply:
no it works efficiently...keeping higher limit sometimes lower performance